### PR TITLE
feat: remove auth file in dataset copy process

### PR DESCRIPTION
**Pull request by anda-ren**
## Description
- remove auth file in dataset copy process
- make storage service configurable by user in system setting
- storage services configured by user could automatically matche with the uris in dataset index
- change uri pattern from `schema://username:password@host@bucket/path` to standered URI which is `schema://username:password@host/path`
- change related docs
- remove `authName` in api

### Storage setting
image

### Evaluations
- Index only dataset evaluation with uri schema `s3`: http://uri-auto-20221110.pre.intra.starwhale.ai/projects/3/evaluations/20/results
- Index only dataset evaluation with uri schema `minio`: http://uri-auto-20221110.pre.intra.starwhale.ai/projects/3/evaluations/21/results
- Binary type dataset with uris without schema: 
http://uri-auto-20221110.pre.intra.starwhale.ai/projects/3/evaluations/22/results

### Vewers
http://uri-auto-20221110.pre.intra.starwhale.ai/projects/3/datasets/2/versions/4/files
Toggle different versions which has different types of uri inside.
## Modules
- [ ] UI
- [x] Controller
- [ ] Agent
- [x] Client
- [ ] Python-SDK
- [x] Others

## Checklist
- [ ] run code format and lint check
- [x] add unit test
- [x] add necessary doc

### CODE REVIEW

File: client/starwhale/core/dataset/store.py [hunk 1]
```diff
@@ -213,8 +217,8 @@ def from_uri(cls, uri: str, auth_name: str) -> S3Connection:
         """make S3 Connection by uri
 
         uri:
-            - s3://username:password@127.0.0.1:8000@bucket/key
-            - s3://127.0.0.1:8000@bucket/key
+            - s3://username:password@127.0.0.1:8000/bucket/key
+            - s3://127.0.0.1:8000/bucket/key
             - s3://bucket/key
```
[the remark this answers is not in the document]
Comment (anda-ren) [reply client/starwhale/core/dataset/store.py:222]: Not really, I'll remove it

File: docs/docs/guides/config/server_admin.md [hunk 2]
```diff
@@ -40,7 +40,27 @@ resourcePoolSetting:
...
+  - type: minio
+    tokens: 
+      - bucket: starwhale
+        ak: access_key
+        sk: scret_key
+        endpoint: http://10.131.0.1:9000
+        region: local
+        hugeFileThreshold: 10485760
+        hugeFilePartSize: 5242880
+
```
[the remark this answers is not in the document]
Comment (anda-ren) [reply docs/docs/guides/config/server_admin.md:63]: yes

[the remark this answers is not in the document]
Comment (anda-ren) [reply docs/docs/guides/config/server_admin.md:63]: image

File: server/controller/src/main/java/ai/starwhale/mlops/domain/job/split/JobSpliteratorEvaluation.java [hunk 3]
```diff
@@ -56,16 +56,20 @@
...
     private final TaskMapper taskMapper;
-
     private final JobMapper jobMapper;
-
-
     private final StepMapper stepMapper;
-
     private final JobSpecParser jobSpecParser;
+    /**
+     * when task amount exceeds 1000, bach insertion will emit an error
```
[the remark this answers is not in the document]
Comment (anda-ren) [reply server/controller/src/main/java/ai/starwhale/mlops/domain/job/split/JobSpliteratorEvaluation.java]: done

File: server/controller/src/main/java/ai/starwhale/mlops/domain/dataset/objectstore/DsFileGetter.java [hunk 4]
```diff
@@ -21,20 +21,26 @@
...
 import lombok.extern.slf4j.Slf4j;
+import org.jetbrains.annotations.NotNull;
 import org.springframework.stereotype.Service;
+import org.springframework.util.StringUtils;
 
 @Slf4j
 @Service
 public class DsFileGetter {
 
+    static final Set<String> SCHEMA_HTTP = Set.of("http", "https");
```
[the remark this answers is not in the document]
Comment (anda-ren) [reply server/controller/src/main/java/ai/starwhale/mlops/domain/dataset/objectstore/DsFileGetter.java]: done

**Comment by jialeicui** [comment]
LGTM

**Review by jialeicui — APPROVED** [review]

File: client/starwhale/core/dataset/store.py [hunk 5]
```diff
@@ -224,24 +227,14 @@ def from_uri(cls, uri: str, auth_name: str) -> S3Connection:
...
-            endpoint = link_auth.endpoint
-            bucket = _nl[0]
-        elif len(_nl) == 2:
-            endpoint, bucket = _nl
-        elif len(_nl) == 3:
-            _key, endpoint, bucket = _nl
-            access, secret = _key.split(":", 1)
-        else:
-            raise FormatError(netloc)
+        endpoint = r.hostname or link_auth.endpoint
```
Comment (tianweidut) [inline client/starwhale/core/dataset/store.py:235]: endpoint will miss the port.

```ipython

In [12]: from urllib.parse import urlparse

In [13]: r= urlparse("s3://username:password@127.0.0.1:8000/bucket/key")

In [14]: r.hostname
Out[14]: '127.0.0.1'

In [16]: r.port
Out[16]: 8000
```

Comment (tianweidut) [reply client/starwhale/core/dataset/store.py:235]: `endpoint = r.netloc.split("@")[-1] or link_auth.endpoint`

**Pull request merged**